apm-agent-core/src/test/java/co/elastic/apm/agent/bci/methodmatching/TraceMethodInstrumentationTest.java

```diff
@@ -62,35 +63,52 @@ void tearDown() {
 
     @Test
     void testTraceMethod() {
-        traceMe();
+        TestClass.traceMe();
         assertThat(reporter.getTransactions()).hasSize(1);
+        assertThat(reporter.getFirstTransaction().getName().toString()).isEqualTo("TestClass#traceMe");
         assertThat(reporter.getSpans()).hasSize(1);
+        assertThat(reporter.getFirstSpan().getName().toString()).isEqualTo("TestClass#traceMeToo");
     }
 
     @Test
     void testNotMatched_VisibilityModifier() {
-        traceMeNot();
+        TestClass.traceMeNot();
         assertThat(reporter.getTransactions()).isEmpty();
     }
 
     @Test
     void testNotMatched_Parameters() {
-        traceMeNot(false);
+        TestClass.traceMeNot(false);
         assertThat(reporter.getTransactions()).isEmpty();
     }
 
-    // not traced because visibility modifier does not match
-    public void traceMeNot() {
+    // Byte Buddy can't catch exceptions (onThrowable = Throwable.class) during a constructor call:
+    @Test
+    void testNotMatched_Constructor() {
+        new TestExcludeConstructor();
+        assertThat(reporter.getTransactions()).isEmpty();
```

Comment:
Just to have feedback that the test is valid, add here a method invocation on that instance and check that it is instrumented